### ISSUE TITLE
[Bug] SSR does not seem to work on new project

### ISSUE THREAD

**Issue by atulmy** [nozzle/react-static#1066]
### Reporting a bug?
Creating a new project with `react-static create` and selecting `basic`, and building with `yarn build` and serving `yarn serve`, when we see source of `/about` page, it shows source of `/` home page.

### Environment
```
Run and copy the result of:
OS: macOS 10.14.3
CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
Memory: 144.34 MB / 8.00 GB
Shell: 5.3 - /bin/zsh
```

### Steps to Reproduce the problem
As described in _Reporting a bug_ section.

### Expected Behavior
The source should only contain HTML code for that particular route.

### Reproducible Demo
As described in _Reporting a bug_ section.

**Comment by tannerlinsley** [comment]
Did you load into the home page then navigate to the about page? View source shows you the HTML file that was originally loaded, not the current Dom

Tanner Linsley

**Comment by atulmy** [comment]
Nope, refresh while being on About Page and tried to view page source. It reflected Home Page's page source. I resorted to cloning https://github.com/nozzle/nozzle.io and working with that. SSR works fine with that repository. Just deployed it for the portfolio website (required SEO) here: https://wip.studio

**Comment by tannerlinsley** [comment]
On my first attempt to replicate this issue, I get the expected functionality. Do you have any other details or steps needed to replicate this issue?

**Comment by atulmy** [comment]
Perhaps something with my local setup or machine. Anyway got it working with the nozzle.io and will use same repository to bootstrap a project in future. Thanks for the reply though.

**Comment by boennemann** [comment]
I'm seeing the exact same behavior. `npm run build` outputs the very same html file for every route and flashes to the actual content.
Adding a siteRoot breaks it entirely, too.

This might also be causing https://github.com/nozzle/react-static/issues/990, which isn't working for me either.

Started from the basic template, which appears to have more than just this issue. (Had to downgrade webpack, and to upgrade react-helmet to beta).

**Comment by boennemann** [comment]
I'm digging through this for quite some time now. I can't find anything in the build process, as the right data seems to be passed and everything.

My last try was this:

```js
// node.api.js
import ReactDOMServer from 'react-dom/server'
import crypto from 'crypto'

export default () => ({
  beforeRenderToHtml: element => {
    const page = ReactDOMServer.renderToString(element)
    const hash = crypto
      .createHash('md5')
      .update(page)
      .digest('hex')

    console.log('page', hash)

    return element
  },
  beforeDocumentToFile: html => {
    const hash = crypto
      .createHash('md5')
      .update(html)
      .digest('hex')
    console.log('html', hash)

    return html
  },
})
```

Result

```
page d41d8cd98f00b204e9800998ecf8427e
page d41d8cd98f00b204e9800998ecf8427e
page d41d8cd98f00b204e9800998ecf8427e
page d41d8cd98f00b204e9800998ecf8427e 
html b1aac5e9cc07d5fae254123cd5240d66
page d41d8cd98f00b204e9800998ecf8427e
html da9c351d6833908bfcf38456bcdaea03
page 6724253b124f664d1108279944a654b4
page d41d8cd98f00b204e9800998ecf8427e // not sure why this is the same a few times? Could this be the redirect/404 page?
html 71877ebbc89e4635c60ff9734145d71c
html 00622dadb71b9990239d535d8983e996
page 846c20eac1769969f3386ef6d243ff3c
page cc86f9f9b4ec4fc5ab29f8afdbd66bbf
page 9b8c9d97423c42fbd1403013c99a2cc8
page 1f1862d8a0ce974d82b2bb9c6a9e8d75
page 93e9c33b74baf182fa158a12cf1ca481
page 93e2be2bc17fa2798e353f82892d530c
page 0dcff8a879319528dffef9550d384a53
page 80174b655362f2a6f3c6ae6432e3a0eb
page 27a0e8f73840b91f8bfa0dc34a3315c2
html 773d96c32388aadec989b29fa96d2317
html a695fe9b9411ebf6fabdc704ea160309
html 394c22b097ff11b5e5c615d8f7f84790
html ad54b51bef63e8300cd648098e30c5be
html cf46a853f9df51af40006d18d6f8194c
html c97ca13a6f51bc4bd68d17571ce738ab
html 85517fd88ee12fce14a587a1e3aa8988
html 2348490f543e04d558410646eb6a0114
page f95047b88b7e5459cccca3b0f594c993
page c5cedb5667fe1a74a4b45e68370f1489
html a1cc07d7cd7a0124364948b70dea646a
page 66700501f8f8af6c013f26d40832192f
html dab1ee3c69961c8df9127f868abde265
html 4ccfd44c2f4309d11c7345641af73e04
html 0a45c8bcadb618f6c03d9311a312af8f
html 0caa4aa5b4903df5c676727644292a54
html 29dd443829d909a792c6fa52762a7249
html 1d5ada960c52505fed09d9ab37b74132
```

It looks like the right pages are being built … and they're indeed different. I just used the hashes for simplification. I also looked at the raw output and it's the correct one.

It seems like this bug is somewhere shortly before writing the html to disk, but it's too late for me here to continue. Maybe someone else wants to continue or I'll need to continue tomorrow.

**Comment by boennemann** [comment]
Holy smokes – I found it. The story of how two tiny characters can mess up an entire day worth of work …

I was digging and digging, when I finally realized there is a difference between the files inside my dist folder, and the ones `view-source:` is showing.

Turns out the `basic` template passes the `-s` flag to `serve`. The reason for this? I have no idea.

```
      -s, --single                        Rewrite all not-found requests to `index.html`
```

Bingo, that's it.

The `-s` flag is not present for nozzle/nozzle.io, which explains that it started working for @atulmy once he used that as a template.
